mysite/lti/tests.py

```diff
@@ -166,33 +151,40 @@ def test_lti_user(self, mocked):
     def test_lti_user_no_email(self, mocked):
         del self.headers[u'lis_person_contact_email_primary']
         mocked.return_value.is_valid_request.return_value = True
-        self.client.post('/lti/',
-                         data=self.headers,
-                         follow=True)
-        self.assertTrue(LTIUser.objects.filter(consumer='moodle').exists())
+        self.client.post('/lti/', data=self.headers, follow=True)
+        self.assertTrue(LTIUser.objects.filter(lti_consumer=self.lti_consumer).exists())
         self.assertTrue(Role.objects.filter(role=Role.ENROLLED).exists())
-        self.assertNotEqual(LTIUser.objects.get(consumer='moodle').django_user,
+        self.assertNotEqual(LTIUser.objects.get(lti_consumer=self.lti_consumer).django_user,
                             User.objects.get(id=self.user.id))
 
-    def test_lti_user_no_username_no_email(self, mocked):
+    @patch('lti.utils.uuid4')
+    def test_lti_user_no_username_no_email(self, random, mocked):
         """Test for non-existent username field
 
         If there is no username in POST
-        we create user with username==user_id
+        we create user with random username.
         """
+        test_random_username = 'c'*32
+
         del self.headers[u'lis_person_name_full']
         del self.headers[u'lis_person_contact_email_primary']
         mocked.return_value.is_valid_request.return_value = True
-        self.client.post('/lti/',
-                         data=self.headers,
-                         follow=True)
-        self.assertTrue(LTIUser.objects.filter(consumer='moodle').exists())
+        random().hex = test_random_username[:30]
```

Comment:
@z4y4ts 
Yes, this will do what I want - I want to mock `lti.utils.uuid4` to create Django User w/ predefined username.
But I can see naming issue there. So I am going to change argument name from `random` to `mocked_uuid4`.